streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,254 @@
+        Map<String, CreateTopicsRequest.TopicDetails> topics = new HashMap<>();
+        topics.put(internalTopicConfig.name(), topicDetails);
+
+        CreateTopicsRequest createTopicsRequest = new CreateTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        ClientResponse clientResponse = sendRequest(createTopicsRequest.toStruct(), ApiKeys.CREATE_TOPICS, null);
+        CreateTopicsResponse createTopicsResponse = new CreateTopicsResponse(clientResponse.responseBody());
+        if (createTopicsResponse.errors().get(internalTopicConfig.name()).code() > 0) {
+            throw new StreamsException("Could not create topic: " + internalTopicConfig.name() + ". " + createTopicsResponse.errors().get(internalTopicConfig.name()).name());
+        }
+    }
+
+    /**
+     * Delete a topic.
+     *
+     * @param topic
+     */
+    public void deleteTopic(final String topic) {
+        log.debug("Deleting topic {} from ZK in partition assignor.", topic);
+        Set<String> topics = new HashSet();
+        topics.add(topic);
+
+        DeleteTopicsRequest deleteTopicsRequest = new DeleteTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        ClientResponse clientResponse = sendRequest(deleteTopicsRequest.toStruct(), ApiKeys.DELETE_TOPICS, null);
+        DeleteTopicsResponse deleteTopicsResponse = new DeleteTopicsResponse(clientResponse.responseBody());
+        if (deleteTopicsResponse.errors().get(topic).code() > 0) {
+            throw new StreamsException("Could not delete topic: " + topic + ". " + deleteTopicsResponse.errors().get(topic).name());
+        }
+
+    }
+
+    /**
+     * Send a request to kafka broker of this client. Polls the request for a given number of iterations to receive the response.
+     *
+     * @param request
+     * @param apiKeys
+     * @param callback
+     */
+    public ClientResponse sendRequest(final Struct request, final ApiKeys apiKeys, final RequestCompletionHandler callback) {
+
+        Node brokerNode = kafkaClient.leastLoadedNode(new SystemTime().milliseconds());
+        String brokerId = Integer.toString(brokerNode.id());
+
+        SystemTime systemTime = new SystemTime();
+
+        RequestSend send = new RequestSend(brokerId,
+                kafkaClient.nextRequestHeader(apiKeys),
+                request);
+
+        ClientRequest clientRequest = new ClientRequest(systemTime.milliseconds(), true, send, callback);
+
+
+        final long timeout = systemTime.milliseconds() + MAX_WAIT_TIME_MS;
+        while (!kafkaClient.ready(brokerNode, systemTime.milliseconds()) && systemTime.milliseconds() < timeout) {
+            kafkaClient.poll(streamsConfig.getLong(StreamsConfig.POLL_MS_CONFIG), systemTime.milliseconds());
+        }
+        if (!kafkaClient.ready(brokerNode, systemTime.milliseconds())) {
+            throw new StreamsException("Request timeout.");
```

Comment:
Probably need a more descriptive message, i.e.,
"Timed out waiting for node=" + brokerNode + " to become available"